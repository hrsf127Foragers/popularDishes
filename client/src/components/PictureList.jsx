import React from 'react';
import styles from './PictureListStyle.css'

const PictureList = ({name, image}) => (

      <div className={styles.mySlides}>
        <img className={styles.photo} src= {image} alt="FOOD1"></img>
        <br></br>
        <div className={styles.dishName}>{name}</div>
        <div className={styles.photoReview}>11 photos - 10 reviews</div>
      </div>
);

export default PictureList;