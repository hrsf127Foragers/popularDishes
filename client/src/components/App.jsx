import PictureList from './PictureList.jsx';
import PopularDishList from './PopularDishList.jsx';
// import ImageSlides from './ImageSlides.jsx';
import ReviewsList from './ReviewsList.jsx';
import Module from './Module.jsx';
import Arrow from './Arrow.jsx';
import axios from 'axios';
import React from 'react';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      popular_dishes: [],
      reviews: [],
      pictures: [],
      restaurant: '',
      currentImageIndex: 0
    }
    // this.previousSlide = this.previousSlide.bind(this);
    // this.nextSlide = this.nextSlide.bind(this);

  }

  // previousSlide() {
  //   const lastIdx = imgUrl.length-1;
  //   const {currentImageIndex} = this.state;
  //   const shouldResetIdx = currentImageIndex === 0;
  //   const idx = shouldResetIdx ? lastIdx: currentImageIndex-1;
  //   this.setState({
  //     currentImageIndex: idx
  //   });
  // }

  // nextSlide(){
  // const lastIdx = imgUrl.length - 1;
  // const { currentImageIndex } = this.state;
  // const shouldResetIdx = currentImageIndex === lastIdx;
  // const idx = shouldResetIdx ? 0: currentImageIndex + 1;
  // this.setState({
  //   currentImageIndex: idx
  // });
  // }

  render() {
    return (
      <div>
        {/* <h3>Popular Dishes</h3>
        <h5> View Full Menu </h5> */}
        {/* <div className="carousel">
          <Arrow
            direction="left"
            clickFunction={this.previousSlide}
            glyph="&#9664;" />

          <Arrow
            direction="right"
            clickFunction={this.nextSlide}
            glyph="&#9654;" />
          <ImageSlides url={imgUrl} />
          {/* <PictureList /> */}

        {/* </div> */}
        <br></br>
          <PictureList />
        <PopularDishList/>
        <ReviewsList/>
      </div>
    )
  }
}

export default App;