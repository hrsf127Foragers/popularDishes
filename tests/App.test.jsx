import React from 'react';
import { shallow, mount, render } from 'enzyme';
import App from '../client/src/Components/App.jsx'
import PictureList from '../client/src/Components/PictureList.jsx'

// describe('App', () => {
//   it('it should be true', () => {
//     const boo = true;
//     expect(boo).toBe (true);
//   });
// });

describe('App', () => {
  it('checks if App component exist', () => {
    const wrapper = shallow(<App/>);
    expect(wrapper).toExist();

  });
  it('should render correctly with no props', () => {
    const wrapper = shallow(<App/>);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render text correctly with passing in props', () => {
    const texts = ['one', 'two']
    const wrapper = shallow(<App numText={texts}/>);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<PictureList /> rendering', () => {
  it('should render <div>', () => {
    const wrapper = shallow(<PictureList/>);
    expect(wrapper.find('div')).toHaveLength(17);
  });

})