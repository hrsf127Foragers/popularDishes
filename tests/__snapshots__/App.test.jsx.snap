// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly with no props 1`] = `
<div>
  <br />
  <PictureList />
  <PopularDishList />
  <ReviewsList />
</div>
`;

exports[`App should render text correctly with passing in props 1`] = `
<div>
  <br />
  <PictureList />
  <PopularDishList />
  <ReviewsList />
</div>
`;
